pennylane_lightning_gpu/src/bindings/Bindings.cpp

```diff
@@ -85,6 +88,43 @@ void StateVectorCudaManaged_class_bindings(py::module &m) {
             return new StateVectorCudaManaged<PrecisionT>(
                 data_ptr, static_cast<std::size_t>(arr.size()));
         }))
+        .def(
+            "SetBasisState",
+            [](StateVectorCudaManaged<PrecisionT> &sv, const size_t index,
+               const bool use_async) {
+                std::complex<PrecisionT> value = {1, 0};
+                sv.setState(value, index, use_async);
+            },
+            "Create basis state on GPU.")
+        .def(
+            "SetStateVector",
+            [](StateVectorCudaManaged<PrecisionT> &sv,
+               const np_arr_sparse_ind &indices, const np_arr_c &state,
+               const bool use_async) {
+                using index_type = typename std::conditional<
+                    std::is_same<ParamT, float>::value, int32_t, int64_t>::type;
+
+                const auto state_buffer = state.request();
+                std::vector<std::complex<PrecisionT>> state_;
+                if (state_buffer.size) {
+                    const auto state_ptr =
+                        static_cast<const std::complex<PrecisionT> *>(
+                            state_buffer.ptr);
+                    state_ = std::vector<std::complex<PrecisionT>>{
+                        state_ptr, state_ptr + state_buffer.size};
+                }
```

Comment:
do we need to create a copy from the numpy buffer data? Can we just read it in directory from the pointer?